Mixmatch/streetview_dataset/parse_TFRecord.ipynb

```diff
@@ -0,0 +1,221 @@
+    "import tensorflow as tf\n",
+    "import matplotlib.pyplot as plt\n",
+    "import IPython.display as display\n",
+    "import itertools\n",
+    "import os  # used for directory operations\n",
+    "import io\n",
+    "from PIL import Image  # used to read images from directory\n",
+    "\n",
+    "tf.enable_eager_execution()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 21,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def read_TFRecord(file_path):\n",
+    "    raw_image_dataset = tf.data.TFRecordDataset(file_path)\n",
+    "\n",
+    "    # Create a dictionary describing the features.\n",
+    "    image_feature_description = {\n",
+    "        'image/source_id': tf.io.FixedLenFeature([], tf.string),\n",
+    "        'image/object/bbox/confidence': tf.io.VarLenFeature(tf.float32),\n",
+    "        'image/object/bbox/xmin': tf.io.VarLenFeature(tf.float32),\n",
+    "        'image/object/bbox/ymin': tf.io.VarLenFeature(tf.float32),\n",
+    "        'image/object/bbox/xmax': tf.io.VarLenFeature(tf.float32),\n",
+    "        'image/object/bbox/ymax': tf.io.VarLenFeature(tf.float32),\n",
+    "    }\n",
+    "\n",
+    "    def _parse_image_function(example_proto):\n",
+    "      # Parse the input tf.Example proto using the dictionary above.\n",
+    "      return tf.io.parse_single_example(example_proto, image_feature_description)\n",
+    "\n",
+    "    parsed_image_dataset = raw_image_dataset.map(_parse_image_function)\n",
+    "    return parsed_image_dataset\n",
+    "\n",
+    "def parse_labels(image_features):\n",
+    "    img_name = str(image_features['image/source_id'].numpy()).split('/')[-1][:-1]\n",
+    "    confidence = tf.sparse_tensor_to_dense(image_features['image/object/bbox/confidence'], default_value=0).numpy()\n",
+    "    xmin = tf.sparse_tensor_to_dense(image_features['image/object/bbox/xmin'], default_value=0).numpy()\n",
+    "    ymin = tf.sparse_tensor_to_dense(image_features['image/object/bbox/ymin'], default_value=0).numpy()\n",
+    "    xmax = tf.sparse_tensor_to_dense(image_features['image/object/bbox/xmax'], default_value=0).numpy()\n",
+    "    ymax = tf.sparse_tensor_to_dense(image_features['image/object/bbox/ymax'], default_value=0).numpy()\n",
+    "    \n",
+    "    bbox = np.vstack((xmin, ymin, xmax, ymax)) # Left, Top, Right, Bottom\n",
+    "    \n",
+    "    return img_name, confidence, bbox\n",
+    "\n",
+    "def img2example(img, label):\n",
+    "    imgByteArr = io.BytesIO()\n",
+    "    img.save(imgByteArr, format='JPEG')\n",
+    "    imgByteArr = imgByteArr.getvalue()\n",
+    "\n",
+    "    example = tf.train.Example(features=tf.train.Features(feature={\n",
+    "        \"image\": tf.train.Feature(bytes_list=tf.train.BytesList(value=[imgByteArr])),\n",
+    "        \"label\": tf.train.Feature(int64_list=tf.train.Int64List(value=[label]))}))\n",
+    "    \n",
+    "    return example"
+   ]
+  },
+  {
+   "cell_type": "code",
```

Comment:
Why not put all function definitions in one cell?

Reply:
The later part is specific for streetview_v1 dataset. So I want to separate general functions and dataset specific functions. I've added some notes on the notebook file to make it more distinct.